### ISSUE TITLE
[Bug] Using AddOnSkins exposes an overlap with the RIO side frame and PGF

### ISSUE THREAD

**Issue by CDAGaming** [0xbs/premade-groups-filter#61]
When using AddOnSkins (Which skins PGF to be transparent and to look more like ElvUI), it shows that the PGF overlay overlaps over RaiderIO.

Edit: This ussue is oresent without AddonSkins but is less noticable due to PGF being non transparent by default.

**Comment by 0xbs** [comment]
Thank your for the report. To be honest, I am not sure if I can invest much time into this issue. There has always been some kind of overlap issue and it is hard to get it right because each addon like to place itself into a different layer of the WoW UI. Additionally, raider.io is slowly getting replaced by Blizzards own score system. I also generally do not support interface skins as there are so many of them and they manipulate quite a lot to make the UI look different causing all sort of side effects (as in #53).

**Comment by CDAGaming** [comment]
The thing is though that this issue occurs by default, but isnt noticable without AddonSkins since by default PGF isnt transparent.

As for RIO, blizz has only replaced this for m+, with no plans to do so for rating. Due to this, I still believe this issue is warranted since rio continues to be used by many players

**Comment by 0xbs** [comment]
Okay I will see if I can do anything when the next change is up. Could you post a screenshot with the issue?

**Comment by CDAGaming** [comment]
Sure thing! Here you go!

![image](https://user-images.githubusercontent.com/16328918/130121500-139d1f8c-f9d6-48ca-9ab8-b558276b1555.png)

**Comment by 0xbs** [comment]
Ah I see! Actually the two windows do not intersect, but PGF is simply above the Raider.IO frame. I could of course shift PGF to a lower UI layer so that it is behind the RIO frame. But then you would not be able to use PGF. The solution is to click the small lock item in the top right of the PGF window. After you unlocked the PGF window, you can move it (by grabbing the title) to a different position, e.g. to the right so that you can see the RIO frame.

**Comment by CDAGaming** [comment]
@0xbs I am aware that you can move it, but that isn't necessarily the point of making this issue. More so, that perhaps there could either be a way to auto-offset the locked position or some other solution. As it is now, the default settings of PGF overlap the RIO side panel (And without addonskins, it is impossible to tell from a normal user's perspective that there is a frame overlap).

**Comment by 0xbs** [comment]
Well I could of course check if the RIO addon is installed, probably also check if the RIO frame is present next to the dungeon finder and in this case trying to bind it to the right of the RIO window. But if I do that, the logic inside my addon is now more complex and also depends even more on RIO. Whenever they release a new version, my addon could break. I already had problems in the past when RIO changed the way you could query the rating of players. Heck, they even had two versions for some weeks: the one from CurseForge and the one from the RIO Client that had completely different APIs. I had to support both versions. I also found it very hard communicating with the developers when trying to sync our keywords in advance. As I think PGF is a year older than RIO, why not open a bug report at RIO so that they move their frame out of the way whenever PGF is open?

**Comment by CDAGaming** [comment]
Done, and CC'd!

**Comment by YoungP036** [comment]
I might be misunderstanding, but what is wrong with unlocking and moving the premade groups filter frame?

**Comment by CDAGaming** [comment]
> I might be misunderstanding, but what is wrong with unlocking and moving the premade groups filter frame?

```
I am aware that you can move it, but that isn't necessarily the point of making this issue. More so, that perhaps there could either be a way to auto-offset the locked position or some other solution. As it is now, the default settings of PGF overlap the RIO side panel (And without addonskins, it is impossible to tell from a normal user's perspective that there is a frame overlap).

Additionally, with moving the panel, there have been cases of the position not being saved as well and having it revert, making it not the overall best solution to this type of issue.```

TL;DR: Multiple issues, mainly that doing that is a bandaid rather then fixing the actual issue (Which as a software developer, I royally hate when a problem is bandaided and then never resolved until months/years later). _The issue however was resolved in an upcoming RIO patch on their end for this._

**Comment by 0xbs** [comment]
Moving the frame is not a bandaid, it's a feature requested by many players for multiple reasons.

But I am happy to see that the issue is being fixed on the rio side which is also the better place in my opinion.

**Comment by CDAGaming** [comment]
While true, in this case (Where two frames are covering each other), moving the frame is a _workaround_ and not a permanent solution to the problem at hand (Where data in a default setting is hidden from the user without knowledge). I had no clue the RIO frame was even under PGF's until I found it when AddOnSkins made PGF transparent, showing the issue.

In any case, this is resolved on RIO's end.
